storage/src/archive/storage.rs

```diff
@@ -270,7 +270,7 @@ impl<T: Translator, K: Array, B: Blob, E: Storage<B> + Metrics> Archive<T, K, B,
 
         // Check for existing index
         if self.indices.contains_key(&index) {
-            return Err(Error::DuplicateIndex);
+            return Ok(());
```

Comment:
It is very common to get data from multiple places (and to insert it). This behavior makes the DX cleaner.

Reply:
I suppose I could revert this and add a `put_duplicate` (or something) function that returns all errors other than this one.